crates/bevy_pbr/src/light.rs

```diff
@@ -568,19 +784,135 @@ pub(crate) fn assign_lights_to_clusters(
 
     let light_count = lights.len();
     let mut global_lights_set = HashSet::with_capacity(light_count);
-    for (view_entity, view_transform, camera, frustum, mut clusters) in views.iter_mut() {
+    for (view_entity, view_transform, camera, frustum, config, mut clusters) in views.iter_mut() {
+        if matches!(config, ClusterConfig::None) {
+            commands.entity(view_entity).remove::<VisiblePointLights>();
+            continue;
+        }
+
         let view_transform = view_transform.compute_matrix();
         let inverse_view_transform = view_transform.inverse();
-        let cluster_count = clusters.aabbs.len();
         let is_orthographic = camera.projection_matrix.w_axis.w == 1.0;
+
+        let screen_size_u32 = camera.target.get_physical_size(&windows, &images);
+        if screen_size_u32.is_none() {
+            continue;
+        }
+        let screen_size_u32 = screen_size_u32.unwrap();
+        let mut cluster_dimensions = config.dimensions_for_screen_size(screen_size_u32);
+
+        let far_z = match config.far_z_mode() {
+            ClusterFarZMode::CameraFarPlane => camera.far,
+            ClusterFarZMode::MaxLightRange => lights
+                .iter()
+                .map(|light| {
+                    (inverse_view_transform * light.translation.extend(1.0)).z * -1.0 + light.range
+                })
+                .reduce(f32::max)
+                .unwrap_or(0.0),
+            ClusterFarZMode::Constant(far) => far,
+        };
+        let first_slice_depth = match cluster_dimensions.z {
+            1 => config.first_slice_depth().max(far_z),
+            _ => config.first_slice_depth(),
+        };
+        let far_z = far_z.max(first_slice_depth);
```

Comment:
```suggestion
        // NOTE: Ensure the far_z is at least as far as the first_depth_slice to avoid clustering problems.
        let far_z = far_z.max(first_slice_depth);
```